### PR TITLE
bcda-837 Encryption walkthrough including Swagger and screenshots

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-837](https://jira.cms.gov/browse/BCDA-837)
The research team will be showing an encryption demonstration to the alpha ACO's.  This document will be shared with them afterward for reference.

### Proposed changes:
* Add a `doc` subdirectory where we can put this (and future) documentation
* Add a `doc/images` subdirectory for screenshots

### Acceptance Validation
screen shot 2019-02-03 at 10 14 28 pm

### Feedback Requested
(navigate [here](https://github.com/CMSgov/bcda-app/blob/dhg/BCDA-837/doc/DECRYPTION_EXAMPLE.md) to see a full preview of the documentation)
* Tone: does the text meet our desired tone for communicating with users, not only for the immediate purpose but also for later audiences?
* Correctness: 
   - Can you follow the instructions and end up with a decrypted file?  
   - Do the links work?
   - Is the encryption vocabulary ambiguous, confusing, or incorrect?
   - What is misspelled?
* Improvements: 
   - Do the screenshots include too much context?  
   - Are the screenshots legible?  
   - Should the information be presented in a different order?
* Utility: is there enough new value in this documentation to warrant maintaining it in addition to `ENCRYPTION.md`?
* Project layout: 
    - Is it worth creating this new `doc` subdirectory?

### CODE REVIEW

File: doc/DECRYPTION_EXAMPLE.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+- Open the data file section in Swagger (click `/api/v1/jobs/{jobID}/{filename}`)
+- Paste the job ID and filename into the appropriate boxes
+- Click `Execute`
+
+<img src="images/decrypt_demo_08.png" alt="Get data file (Swagger)" width="600" />
+
+- Click the `Download file` link that appeared in the response section.  Note that a large sample file may take a while to download.
+
+<img src="images/decrypt_demo_09.png" alt="Download file" width="600" />
```
Comment (knollfear) [inline doc/DECRYPTION_EXAMPLE.md]: Does this image contain a token?   Should we obfuscate it a bit?

File: doc/DECRYPTION_EXAMPLE.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+<img src="images/decrypt_demo_14.png" alt="Parsing decrypted file contents" width="900" />
+
+### Reading further
+* [Encryption documentation](../ENCRYPTION.md)
+* [API documentation](../API.md)
+
+### Troubleshooting
+#### Authentication problems in Swagger
+- Has your token expired?  [Ask the support team](https://groups.google.com/forum/#!forum/bc-api) for a new token.
+- Is the lock on the `Authorize` icon not closed?  Click it again, and after pasting your token in the `api_key` box, make sure to click the `Authorize` button.
```
Comment (knollfear) [inline doc/DECRYPTION_EXAMPLE.md]: I have clicked the logout button a few times by mistake, might be worth noting that.

File: doc/DECRYPTION_EXAMPLE.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+- Are there any spaces or newlines in your token?  Remove them and paste it as a single line.
+
+#### Python not installed
+- Is this your first time running Python on your system?  You might be interested in [this Windows installation guide](https://www.howtogeek.com/197947/how-to-install-python-on-windows/)
```
Comment (knollfear) [inline doc/DECRYPTION_EXAMPLE.md]: python --version as a validation

**Comment by epfrankle** [comment]
Found a few more things in link checks:

* Is requirements.txt supposed to link to just that single line of code?
* Should we be linking out to this branch? https://github.com/CMSgov/bcda-app/blob/dhg/BCDA-837/API.md
* Should https://sandbox.bcda.cms.gov/api/v1/swagger/ resolve? (I'm not on VPN at present and currently it just spins)

**Comment by dhgreene** [comment]
> * Is requirements.txt supposed to link to just that single line of code?
Yes

> * Should we be linking out to this branch? https://github.com/CMSgov/bcda-app/blob/dhg/BCDA-837/API.md
This should be a relative link.  When PR #160 is approved, this document will be at https://github.com/CMSgov/bcda-app/doc/DECRYPTION_EXAMPLE.md and will link to https://github.com/CMSgov/bcda-app/API.md.

> * Should https://sandbox.bcda.cms.gov/api/v1/swagger/ resolve? (I'm not on VPN at present and currently it just spins)
It won't work until we publish R14 to sandbox, and then only for those with whitelisted IP's or those on VPN.

Hopefully this PR is approved today, and R14 is published to sandbox today!

File: doc/DECRYPTION_EXAMPLE.md (hunk 1 above)
Comment (dhgreene) [reply doc/DECRYPTION_EXAMPLE.md]: Yes, a local token that's never been near one of our servers.  I don't think obfuscation improves our security, and I think it detracts somewhat from readability.

File: doc/DECRYPTION_EXAMPLE.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+#### [https://sandbox.bcda.cms.gov/api/v1/swagger](https://sandbox.bcda.cms.gov/api/v1/swagger/)
+<img src="images/decrypt_demo_01.png" alt="Screenshot of swagger" width="400" />
+
+Let's provide our token to get full access to the API.
+
+<img src="images/decrypt_demo_02.png" alt="Authorization dialog" width="350" />
+
+- Click the `Authorize` button
+- Paste your `token` in the `api_key` box
```
[the remark this answers is not in the document]
Comment (dhgreene) [reply doc/DECRYPTION_EXAMPLE.md]: That's very interesting that "bearer " is not needed locally.  As soon as `dev` settles down I'll test it out and update the doc.

**Pull request merged**